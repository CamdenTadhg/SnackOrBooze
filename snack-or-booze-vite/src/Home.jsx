import React from "react";
import {Link} from 'react-router-dom';
import { Card, CardBody, CardTitle, ListGroup, ListGroupItem} from "reactstrap";
import 'bootstrap/dist/css/bootstrap.min.css';

//renders home page of app
function Home({snacks, drinks}) {
  return (
    <section className="col-md-8">
      <Card>
        <CardBody className="text-center">
          <CardTitle>
            <h5 className="font-weight-bold">
              Welcome to Silicon Valley's premier dive cafe!
            </h5>
          </CardTitle>
            <ListGroup>
              <Link to={'/snacks'}>
                <ListGroupItem>Snacks: {snacks.length}</ListGroupItem>
              </Link>
              <Link to={'/drinks'}>
                <ListGroupItem>Drinks: {drinks.length}</ListGroupItem>
              </Link>
            </ListGroup>
        </CardBody>
      </Card>
    </section>
  );
}

export default Home;