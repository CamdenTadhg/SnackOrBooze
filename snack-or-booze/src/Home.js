import React from "react";
import {Link} from 'react-router-dom';
import { Card, CardBody, CardTitle, CardText, ListGroup, ListGroupItem} from "reactstrap";

function Home({snacks, drinks}) {
  return (
    <section className="col-md-8">
      <Card>
        <CardBody className="text-center">
          <CardTitle>
              Welcome to Silicon Valley's premier dive cafe!
          </CardTitle>
            <ListGroup>
              <Link to={'/snacks'}>
                <ListGroupItem>Snacks: {snacks.length}</ListGroupItem>
              </Link>
              <Link to={'/drinks'}>
                <ListGroupItem>Drinks: {drinks.length}</ListGroupItem>
              </Link>
            </ListGroup>
        </CardBody>
      </Card>
    </section>
  );
}

export default Home;
