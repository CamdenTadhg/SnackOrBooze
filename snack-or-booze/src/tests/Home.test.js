import React from 'react';
import {render} from '@testing-library/react';
import Home from "../Home";
import {MemoryRouter} from 'react-router-dom';

const snacks = ['nachos', 'hummus', 'arugula-and-walnut-salad', 'cookie', 'chipscheese'];
const drinks = ['martini', 'negroni', 'gin-and-tonic', 'diet-coke', 'kir', 'manhattan', 'mary'];

it("renders without crashing", function() {
    render(
    <MemoryRouter>
        <Home snacks={snacks} drinks={drinks}/>
    </MemoryRouter>);
});

it("matches snapshot", function() {
    const {asFragment} = render(
        <MemoryRouter>
            <Home snacks={snacks} drinks={drinks}/>
        </MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
});

it('displays the correct content', function() {

    const {getByText} = render(
        <MemoryRouter>
            <Home snacks={snacks} drinks={drinks}/>
        </MemoryRouter>);
    expect(getByText('Snacks: 5', {exact: false})).toBeInTheDocument();
    expect(getByText('Drinks: 7', {exact: false})).toBeInTheDocument();
});