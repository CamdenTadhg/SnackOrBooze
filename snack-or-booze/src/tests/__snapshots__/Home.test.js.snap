// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="col-md-8"
  >
    <div
      class="card"
    >
      <div
        class="text-center card-body"
      >
        <h5
          class="card-title"
        >
          Welcome to Silicon Valley's premier dive cafe!
        </h5>
        <ul
          class="list-group"
        >
          <a
            href="/snacks"
          >
            <li
              class="list-group-item"
            >
              Snacks: 5
            </li>
          </a>
          <a
            href="/drinks"
          >
            <li
              class="list-group-item"
            >
              Drinks: 7
            </li>
          </a>
        </ul>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
